extensions-core/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcStructConverter.java

```diff
@@ -139,18 +155,46 @@ private static Object convertPrimitive(TypeDescription fieldDescription, Writabl
    * primitive types will be extracted into an ingestion friendly state (e.g. 'int' and 'long'). Finally,
    * if a field is not present, this method will return null.
    *
-   * Note: "Union" types are not currently supported and will be returned as null
+   * Note: "Union" types are not currently supported and will be returned as null. Additionally, this method
+   * has a cache of field names to field index that is ONLY valid for the root level {@link OrcStruct}, and should
```

Comment:
This design is not foolproof enough- it's risky/errorprone, and a bit obtuse to read, because two methods called `convertField` with slightly different arguments have very different semantics. This one should be renamed to `convertRootField` and the javadoc should call out the restriction prominently, rather than in a side node.

Reply:
Agree, will fix :+1: